### DISCUSSION TITLE
Made Smart Publish and Revert Workflow

### DISCUSSION THREAD

**Discussion by anantakumarghosh** [wrappid/.github#2] [Show and tell]
This workflow triggers 4 workflows sequentially
All workflows are dependent in chaining way. 
If current workflow runs successfully, then only next workflow will be triggered. 
If any of the workflow fails, published packages by the workflow run are deleted, and version gets decremented
Here is a screenshot :)
![Screenshot 2023-04-28 at 17-58-22 Smart Publish and Revert · wrappid_workflow-mgmt@79d9097](https://user-images.githubusercontent.com/61864488/235147514-38425773-c83d-438a-a085-f033db0f3870.png)

**Comment by samhere17** [comment]
This is a good one... will be very helpful.

Can we extend it to pass parameters to cherry-pick the repos we want to publish?
@anantakumarghosh let me know your thoughts.

@techoneel @sumanta-m

**Reply by techoneel** [reply]
Well cherry pick the repos will be a great idea.

But in terms of sequencing I'd suggest the below structure:

- [ ] styles
  - [ ]  sc-web
  - [ ]  sc-mobile
    - [ ] core

Explanation:
- Publish **styles**package
  If success
- Publish **sc-web** & **sc-mobile** package together
  If success
- Publish **core** package

**Reply by anantakumarghosh** [reply]
If you are thinking of a dynamic approach, based on my current knowledge, I don't know yet. As for passing repos as arguments, **_I think_** it should be possible, number of repos will need to be maintained. Current workflow can do only 4 repos, not more or less. It is possible to make separate workflows for _**specified**_ number of repos that will take repo name as argument.

Or if I'm not understanding you, lets do a meet after my internals are over and when you are available.

**Reply by samhere17** [reply]
@techoneel I agree and I think @anantakumarghosh has followed steps as mentioned by you.

@anantakumarghosh You understood correct. I suggest to pass repo name(s) as argument. It should be a list of repos with dependencies. Lets discuss in details when we are available next.

**Comment by anantakumarghosh** [comment]
Smart publish is not used anymore. Fast Publish is used.
Hence, closing this discussion.
